base/shell/cmd/console.c

```diff
@@ -329,4 +329,39 @@ BOOL ConSetScreenColor(HANDLE hOutput, WORD wColor, BOOL bFill)
 }
 #endif
 
+#include <cjkcode.h>
+#include "wcwidth.c"
+
+SIZE_T ConGetTextWidthW(PCWSTR pszText)
+{
+    SIZE_T ich, cxTextWidth;
+
+    if (!IsCJKCodePage(OutputCodePage))
+        return _tcslen(pszText);
+
+    for (ich = cxTextWidth = 0; pszText[ich]; ++ich)
+    {
+        if (mk_wcwidth_cjk(pszText[ich]) == 2)
```

Comment:
A small remark when looking at what was already done in consrv:
in https://git.reactos.org/?p=reactos.git;a=blob;f=win32ss/user/winsrv/consrv/include/conio.h;hb=dabe7fba461a3d1a48932004ac529d6609d94b38#l342 we have this macro:
```c
// NOTE: The check against 0x80 is to avoid calling the helper function
// for characters that we already know are not full-width.
#define IS_FULL_WIDTH(wch)  \
    (((USHORT)(wch) >= 0x0080) && (mk_wcwidth_cjk(wch) == 2))
```
Would it be useful to use such construct there too?

Reply:
Yeah, but i have no PC tonight to strengthen the memory.

Reply:
:thinking:

Reply:
Now I have my PC.

Reply:
Done.